src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -310,33 +331,59 @@ internal override sealed bool TryReadAsObject(ref Utf8JsonReader reader, JsonSer
         }
 
         /// <summary>
-        /// Overridden by the nullable converter to prevent boxing of values by the JIT.
+        /// Performance optimization. Overridden by the nullable converter to prevent boxing of values by the JIT.
+        /// Although the JIT added support to detect this IL pattern (box+isinst+br), it still manifests in TryWrite().
+        /// </summary>
+        internal virtual bool IsNull(in T value) => value is null;
+
+        /// <summary>
+        /// Performance optimization to determine if IsNull() above should be called.
         /// </summary>
-        internal virtual bool IsNull(in T value) => value == null;
+        /// <returns></returns>
+        internal virtual bool IsNullableOfT() => false;
 
+#if NET6_0_OR_GREATER
+        [MethodImpl(MethodImplOptions.AggressiveOptimization)]
```

Comment:
I was not able to squeeze any performance gains when applying aggressive optimizations to this method in the past.